### ISSUE TITLE
Some feature requests

### ISSUE THREAD

**Issue by 1024mb** [axcore/tartube#233]
## Inform of the actual error code instead of just output Error in the Status column

Can you please change the message from just "Error" to something like "Error 404" / "Error: Not Found" or "Error 403" / "Error: Forbidden"? It would be great if we could see which error we get with each video.

## Delete duplicate URLs instead of keeping them in the text box

In Classic Mode when we add a list of URLS that contain some URLs already added to the queue Tartube only adds the new ones and the rest stays in the text box, can you please add an option or change it so the duplicated entries are deleted from the text box?

## Add an option to show a Title column in Classic Mode

Just like in the original yt-dlg, can we get a new column that shows first the URL and then when it is processed the filename of the output file instead of only the source URL?
This is handy to detect bad/wrong downloads. I have in my filename pattern the width of the video so when I see that a vid is or was downloaded with a wrong resolution I know I have to look into it, the same when I see that instead of the width it says NA (this has happened sometimes when there has been too many requests to the website, after waiting some minutes it downloads fine again) but in order to detect those problems I need to read/see the filename of the output file which, at least in classic mode, I can't.

## When a file is already downloaded change the URL to the filename

When I adding a new URL and the video has already been downloaded replace the URL with the filename in the Videos tab like with the vids that do download. Also fill in the size info (for the urls that have been already downloaded) with what ytdlc gives, i.e.:

```
[info] Writing video description to: X:/tartube/Unsorted Videos/vid1 (*ID*) [720p].description
[download] X:/tartube/Unsorted Videos/vid1 (*ID*) [720p].mp4 has already been downloaded
[download] 100% of 23.13MiB
```

But the size in the Videos tab is unknown.

## Add the warning icon to the simplest view of the Videos tab

If I use the simplest view (the list view) on the Videos tab I can only see two types of icons at the left of the videos, the warning icon is never shown. Can you please add it to that view?

## Delete all but errored entries in Classic Mode

To finish this endless request list I would like to request an option to delete all the entries but the ones that had an error in the Classic Mode.

Thanks!

**Comment by axcore** [comment]
>Can you please change the message from just "Error" to something like "Error 404" / "Error: Not Found" or "Error 403" / "Error: Forbidden"? It would be great if we could see which error we get with each video.

At the moment, there is no way to connect an error message with an individual video. I have added feature requests to both youtube-dl and youtube-dlc, but neither of them have implemented it.

At the moment, you are seeing the most-significant result: if there are 999 successful downloads and 1 warning, you will see WARNING, and if there are 999 successful downloads and one error, you will see ERROR.

>Also fill in the size info (for the urls that have been already downloaded) with what ytdlc gives

youtube-dl(c) supplies the size of the file(s) it is downloading, but there might be several files for one video, and none of them match the size of the final video. Tartube waits until the final video actually exists on the filesystem, before it fetches the size. That's why when you **Check** videos, the file size is marked as **unknown**.

**Comment by axcore** [comment]
>When I adding a new URL and the video has already been downloaded replace the URL with the filename in the Videos tab 

I don't understand this.

**Comment by axcore** [comment]
Ok, I have implemented some of these requests in v2.3.045. The updated code is only on git, so download it from here, and replace the old files in C:\Users\YOURNAME\AppData\Local\Tartube\msys64\home\user\tartube with the new ones.

>Inform of the actual error code instead of just output Error in the Status column
Above, I mentioned that youtube-dl(c) does not tell us which error belongs to which video. Also, there is not enough room in the **Progress** and **Classic Mode** tab to show the error messages. However, I have added the first error/warning message, and the total number of errors/warnings, to the tooltip. Let your mouse hover above a video to see the extended tooltips.

This behaviour can be disabled. **Edit > System preferences... > Windows > Main window**, and deselect **Show errors/warnings in tooltips (but not in the Videos tab)**.

>Delete duplicate URLs instead of keeping them in the text box
Done, but disabled by default. Click **Edit > System preferences... > Windows > Main window**, and select **In the Classic Mode Tab, when adding URLs, remove duplicates rather than retaining them**.

>Add an option to show a Title column in Classic Mode
I didn't want to change the format of the list in the **Classic Mode** tab, because then it won't be the same format as the the list in the **Progress** tab.

But, I have changed the code, so that the video name (if known) is not removed at the end of the download. This should help you identify broken links.

This only applies when you are downloading videos separately. If you're downloading a whole channel or a whole playlist, then the video names are still removed.

>Add the warning icon to the simplest view of the Videos tab
Done.

>Delete all but errored entries in Classic Mode
Done, there is a new **Delete downloaded** button for that.

**Comment by 1024mb** [comment]
Thank you! I didn't think these features were going to be added/developed so fast. You are amazing!

One thing I just saw is that now in Classic Mode when I add duplicated URLs at the same time -by this I mean adding twice the same URL in the same add operation so I have to times one URL in the text box before clicking the "Add URLs button" - Tartube doesn't remove the duplicated URLs, see:

![python3_2021-01-05_04-20-18](https://user-images.githubusercontent.com/9301204/103628685-45eeb700-4f37-11eb-847f-eefca682b80f.png)

**Comment by axcore** [comment]
Thanks, I have fixed it.

**Comment by axcore** [comment]
That fix is in v2.3.072. I think that's everything, but feel free to reopen or start a new issue if not.

**Comment by 1024mb** [comment]
Hey thanks for all your work, I'm finally testing this properly and I've encountered some "bugs".

- If an entry already exists in the destination folder but is a symlink Tartube doesn't fill the "incoming file" column with the filename, of course if it is a regular file it does fills it.
- If the "Download all" button is pressed (this is in the Classic Mode) Tartube starts to re-download all the entries including the already finished ones.

**Comment by axcore** [comment]
Both fixed in v2.3.085.
